library/azure_rm_cdnendpoint.py

```diff
@@ -0,0 +1,623 @@
+
+                    if self.started == False and response['resource_state'] == 'Running':
+                        self.log("Need to stop the CDN endpoint")
+
+                        if not self.check_mode:
+                            self.results = self.stop_cdnendpoint()
+                            self.log("Endpoint stopped")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    elif self.started and response['resource_state'] == 'Stopped':
+                        self.log("Need to start the CDN endpoint")
+
+                        if not self.check_mode:
+                            self.results = self.start_cdnendpoint()
+                            self.log("Endpoint stopped")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    elif self.started is not None:
+                        self.log("Start/Stop not performed due to current resource state")
+                        self.results=response
+                        self.results['changed'] = False
+                        return self.results
+
+                    if self.purge:
+                        self.log("Need to purge endpoint")
+
+                        if not self.check_mode:
+                            self.results = self.purge_cdnendpoint()
+                            self.log("Endpoint purged")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    if update_tags:
+                        to_be_updated = True
+
+                    to_be_updated = to_be_updated or self.check_update(response)
+
+                    if to_be_updated:
+                        self.log("Need to update the CDN endpoint")
+
+                        if not self.check_mode:
+                            self.results = self.update_cdnendpoint()
+                            self.log("Update done")
+
+                        self.results['changed'] = True
+                        return self.results
+
+        elif self.state == 'absent' and response:
+            self.log("Need to delete the CDN endpoint")
+            self.results['changed'] = True
+
+            if not self.check_mode:
+                self.delete_cdnendpoint()
+                self.log("CDN endpoint deleted")
+
+            return self.results
+
+        return self.results
+
+    def create_cdnendpoint(self):
+        '''
+        Creates a Azure CDN endpoint.
+
+        :return: deserialized Azure CDN endpoint instance state dictionary
+        '''
+        self.log("Creating the Azure CDN endpoint instance {0}".format(self.name))
+
+        parameters = Endpoint(
+            origins=[
+                DeepCreatedOrigin(
+                    name=self.origin['name'],
+                    host_name=self.origin['host_name'],
+                    http_port=self.origin['http_port'],
```

Comment:
none check? http_port and https_port is not  required

Reply:
In DeepCreatedOrigin constructor, http_port and https_port are set to None if not provided. In this case, service will set a default value itself(80 for http_port and 443 for https_port).
Should we set the default value in the module or let the service do the work?